File: baekjoon/P13300.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

/**
 * @see <a href="https://www.acmicpc.net/problem/13300">방 배정</a>
 */
public class P13300 {

    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer tokenizer = new StringTokenizer(reader.readLine());

        int[][] grades = new int[2][6];
        int studentNum = Integer.parseInt(tokenizer.nextToken());
        int max = Integer.parseInt(tokenizer.nextToken());

        for (int i = 0; i < studentNum; ++i) {
            tokenizer = new StringTokenizer(reader.readLine());
            int gender = Integer.parseInt(tokenizer.nextToken());
            int grade = Integer.parseInt(tokenizer.nextToken());
            grades[gender][grade - 1]++;
        }

        int minRoomCount = 0;
        for (int i = 0; i < 2; ++i) {
            for (int j = 0; j < 6; ++j) {
                minRoomCount += Math.ceil((double)grades[i][j] / (double)max);
            }
        }

        System.out.println(minRoomCount);
    }

}

File: baekjoon/P13300.cc
#include <cmath>
#include <iostream>

using std::cin;
using std::cout;
using std::ios;

/**
 * @see [방 배정](https://www.acmicpc.net/problem/13300)
 */
int main() {
    ios::sync_with_stdio(false), cin.tie(NULL), cout.tie(NULL);
    int grades[2][6] = {0};
    int student_num, max, gender, grade;
    int min_room_count = 0;

    cin >> student_num >> max;
    for (int i = 0; i < student_num; ++i) {
        cin >> gender >> grade;
        grades[gender][grade - 1]++;
    }

    for (int i = 0; i < 2; ++i) {
        for (int j = 0; j < 6; ++j) {
            min_room_count += ceil(double(grades[i][j]) / double(max));
        }
    }

    cout << min_room_count;
}